### PR TITLE
Implement pagination in gist modal

### LINKED ISSUE #501: Number and overflow-y of personal-gists
The access to your own gists within the vega-editor is nice! 

But:
a) It only shows a limited number of gists. Maybe default to show the first 100 available charts within a persons gist repo?
b) This `div.personal-gist` element has no overflow-y. Now when scrolling down in the overview of gists the `div.load-gist` becomes out of sight.

### PR DESCRIPTION
**Pull request by algomaster99**
Fixes #501 
pagination

### CODE REVIEW

**Comment by domoritz** [comment]
Can you hide pagination if there is only one page?

**Comment by algomaster99** [comment]
@domoritz done

File: src/components/header/gist-modal/renderer.tsx [hunk 1]
```diff
@@ -28,8 +30,9 @@ interface State {
   invalidUrl: boolean;
   latestRevision: boolean;
   loaded: boolean;
+  pages: any;
   personalGist: any;
-  private: boolean;
+  private: string;
```
Comment (domoritz) [inline src/components/header/gist-modal/renderer.tsx]:
```suggestion
  private: GistPrivacy.PUBLIC | GistPrivacy.ALL;
```

**Comment by algomaster99** [comment]
@domoritz 
>  don't let me switch the page, show that you are working e.g. fade out the list or something

I don't get this. What do you mean?

**Comment by domoritz** [comment]
I was thinking that while I am loading a page, I should not be able to switch the page. You can decide to cancel the previous request instead.

**Comment by algomaster99** [comment]
@domoritz What pagination size do you suggest? 10 is too much, I think.

**Comment by domoritz** [comment]
8 to 12

**Comment by algomaster99** [comment]
@domoritz 
load

File: src/components/header/gist-modal/renderer.tsx [hunk 2]
```diff
@@ -58,6 +59,7 @@ class GistModal extends React.PureComponent<Props, State> {
       latestRevision: false,
       loaded: false,
       pages: {},
+      pageLoaded: true,
```
Comment (domoritz) [inline src/components/header/gist-modal/renderer.tsx]: Let's rename this to `loading`.

**Comment by algomaster99** [comment]
@domoritz Can you share the URL of that gist?

**Comment by domoritz** [comment]
Oh odd but okay, let's ignore it.

**Pull request merged**